release/src/main/scripts/run_rc_validation.sh

```diff
@@ -289,283 +290,297 @@ if [[ $confirmation = "y" ]]; then
-  echo "export MOBILE_GAME_GCS_BUCKET=${MOBILE_GAME_GCS_BUCKET}" >> ~/.bashrc
-  echo "export GOOGLE_APPLICATION_CREDENTIALS=${GOOGLE_APPLICATION_CREDENTIALS}" >> ~/.bashrc
-  echo "export RELEASE=${RELEASE}" >> ~/.bashrc
-  echo "export FIXED_WINDOW_DURATION=${FIXED_WINDOW_DURATION}" >> ~/.bashrc
-
-
-  echo "--------------------------Updating ~/.m2/settings.xml-------------------------"
-  cd ~
-  if [[ -d .m2 ]]; then
-    mkdir .m2
-  fi
-  cd .m2
-  if [[ -f ~/.m2/settings.xml ]]; then
-    mv settings.xml settings_backup.xml
-  fi
-  touch settings.xml
-  echo "<settings>" >> settings.xml
-  echo "  <profiles>" >> settings.xml
-  echo "    <profile>" >> settings.xml
-  echo "      <id>release-repo</id>" >> settings.xml
-  echo "      <activation>" >> settings.xml
-  echo "        <activeByDefault>true</activeByDefault>" >> settings.xml
-  echo "      </activation>" >> settings.xml
-  echo "      <repositories>" >> settings.xml
-  echo "        <repository>" >> settings.xml
-  echo "          <id>Release ${RELEASE} RC${RC_NUM}</id>" >> settings.xml
-  echo "          <name>Release ${RELEASE} RC${RC_NUM}</name>" >> settings.xml
-  echo "          <url>${REPO_URL}</url>" >> settings.xml
-  echo "        </repository>" >> settings.xml
-  echo "      </repositories>" >> settings.xml
-  echo "    </profile>" >> settings.xml
-  echo "  </profiles>" >> settings.xml
-  echo "</settings>" >> settings.xml
-
-  echo "----------------------Starting Pubsub Java Injector--------------------------"
-  cd ~/${LOCAL_CLONE_DIR}
-  mvn archetype:generate \
-      -DarchetypeGroupId=org.apache.beam \
-      -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
-      -DarchetypeVersion=${RELEASE} \
-      -DgroupId=org.example \
-      -DartifactId=word-count-beam \
-      -Dversion="0.1" \
-      -Dpackage=org.apache.beam.examples \
-      -DinteractiveMode=false \
-      -DarchetypeCatalog=internal
-
-  cd word-count-beam
-  echo "A new terminal will pop up and start a java top injector."
-  gnome-terminal -x sh -c \
-  "echo '******************************************************';
-   echo '* Running Pubsub Java Injector';
-   echo '******************************************************';
-  mvn compile exec:java -Dexec.mainClass=org.apache.beam.examples.complete.game.injector.Injector \
-  -Dexec.args='${USER_GCP_PROJECT} ${MOBILE_GAME_PUBSUB_TOPIC} none';
-  exec bash"
-
-  echo "[Confirmation Required] Please enter y to confirm injector running:"
-  read confirmation
-  if [[ $confirmation != "y" ]]; then
-    echo "Following tests only can be ran when java injector running."
-    clean_up
-    exit
+  read -a PYTHON_VERSIONS_TO_VALIDATE
+  if [[ -z "$PYTHON_VERSIONS_TO_VALIDATE" ]]; then
+    PYTHON_VERSIONS_TO_VALIDATE=${DEFAULT_PYTHON_VERSIONS_TO_VALIDATE[@]}
   fi
 
-  cd ~/${LOCAL_CLONE_DIR}/
+  for py_version in "${PYTHON_VERSIONS_TO_VALIDATE[@]}"
+  do
+    rm -rf ./beam_env
```

Comment:
I would recommend having python version in env path to make them unique and avoid potential conflict.

Reply:
Right - I think we need to recreate the directory anyway, because virtualenv will not clean up a previously installed artifacts inside a virtualenv directory, if we re-run the script multiple times. That's why I added line 308, and since we create beam_env from scratch everytime, I think there shouldn't be a conflict anymore. Let me know if you think it's still not safe. I can also add a test line to make sure the directory does not exist before we create a new virtualenv.

Reply:
done